src/pybind/mgr/dashboard/frontend/src/app/core/user-info/user-info.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Location } from '@angular/common';
+import { Component, NgZone, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  Validators
+} from '@angular/forms';
+import { Router } from '@angular/router';
+
+import * as _ from 'lodash';
+import { ToastsManager } from 'ng2-toastr/ng2-toastr';
+
+import { AuthService } from '../../shared/api/auth.service';
+import { UserService } from '../../shared/api/user.service';
+import { User } from '../../shared/models/user';
+import { AuthStorageService } from '../../shared/services/auth-storage.service';
+
+@Component({
+  selector: 'cd-user-info',
+  templateUrl: './user-info.component.html',
+  styleUrls: ['./user-info.component.scss']
+})
+export class UserInfoComponent implements OnInit {
+  userForm: FormGroup;
+  username: string;
+
+  constructor(
+    private location: Location,
+    private authStorageService: AuthStorageService,
+    private userService: UserService,
+    public toastr: ToastsManager,
+    private authService: AuthService,
+    private router: Router,
+    private ngZone: NgZone
+  ) {
+    this.userForm = new FormGroup(
+      {
+        username: new FormControl('', {
+          validators: [Validators.required]
+        }),
+        password: new FormControl(),
+        confirmPassword: new FormControl()
+      },
+      {
+        validators: [this.MatchPassword]
+      }
+    );
+  }
+
+  ngOnInit() {
+    this.username = this.authStorageService.get();
+    this.userService.get(this.username).subscribe((res: User) => {
+      this.userForm.patchValue(res);
```

Comment:
`setValue` requires that the object you pass has values for all the `FormControls`.
In this case `res` only has the value for `username,` so it returns an exception if I try to use `setValue`.

With `patchValue` you can pass the value of any number of `FormControls` and it will update just those.